fairlearn/metrics/_metric_frame.py

```diff
@@ -382,19 +386,46 @@ def group_max(self) -> Union[Any, pd.Series, pd.DataFrame]:
         whether control features are present, and whether the metric functions
         were specified as a single callable or a dictionary.
 
+        Parameters
+        ----------
+        errors: {'raise', 'coerce'}, default 'raise'
+            if 'raise', then invalid parsing will raise an exception
+            if 'coerce', then invalid parsing will be set as NaN
+
         Returns
         -------
         typing.Any or pandas.Series or pandas.DataFrame
             The maximum value over sensitive features. The exact type
             follows the table in :attr:`.MetricFrame.overall`.
         """
+        if errors not in _VALID_ERROR_STRING:
+            raise ValueError(_INVALID_ERRORS_VALUE_ERROR_MESSAGE)
+
         if not self.control_levels:
-            result = pd.Series(index=self._by_group.columns, dtype='object')
-            for m in result.index:
-                max_val = self._by_group[m].max()
-                result[m] = max_val
+            if errors == "raise":
+                try:
+                    mf = self._by_group
+                    max_vals = [mf[m].max() for m in mf.columns]
+                    result = pd.Series(max_vals, index=self._by_group.columns, dtype='object')
+                except ValueError:
+                    raise ValueError(_MF_CONTAINS_NON_SCALAR_ERROR_MESSAGE)
```

Comment:
yes, more precisely: 

```suggestion
                except ValueError as e:
                    raise ValueError(_MF_CONTAINS_NON_SCALAR_ERROR_MESSAGE) from e
```